contracts/abstract/Version.sol

```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+abstract contract Version {
+    function getVersion() public view virtual returns (string memory);
```

Comment:
suggestion: encapsulate all the Version behaviour in this contract.

Remove `virtual` to prevent overriding and store the value as a constant.
The advantage of using a static value over an injected one is transparency to users who view the verified contract (i.e. on Etherscan), it'll be visible to them.